packages/react-core/src/filters/tileFeaturesGeometries.js

```diff
@@ -4,36 +4,84 @@ import booleanWithin from '@turf/boolean-within';
 import intersect from '@turf/intersect';
 import transformToTileCoords from '../utils/transformToTileCoords';
 import { TILE_FORMATS } from '../types';
+import transformTileCoordsToWGS84 from '../utils/transformTileCoordsToWGS84';
 
 const GEOMETRY_TYPES = Object.freeze({
   Point: 0,
   LineString: 1,
   Polygon: 2
 });
 
+function processTileFeatureProperties({
+  map,
+  data,
+  startIndex,
+  endIndex,
+  type,
+  bbox,
+  tileFormat,
+  uniqueIdProperty,
+  storeGeometry,
+  geometryIntersection = null
+}) {
+  const tileProps = getPropertiesFromTile(data, startIndex);
+  const uniquePropertyValue = getUniquePropertyValue(tileProps, uniqueIdProperty, map);
+
+  if (uniquePropertyValue && !map.has(uniquePropertyValue)) {
+    let geometry;
+
+    // Only calculate geometry if necessary
+    if (storeGeometry || geometryIntersection) {
+      const { positions } = data;
+      const ringCoordinates = getRingCoordinatesFor(startIndex, endIndex, positions);
+      geometry = getFeatureByType(ringCoordinates, type);
+    }
+
+    // If intersection is required, check before proceeding
+    if (!geometryIntersection || intersects(geometry, geometryIntersection)) {
+      const properties = parseProperties(tileProps);
+
+      // Only save geometry if necessary
+      if (storeGeometry) {
+        properties.geom =
```

Comment:
What would happen if the source contains a column named `geom`, could we have collisions between them?

If the answer is yes, maybe we can name this property as `__internal_tileset_geometry` (or similar), and expose it for its usage in `cloud-native` when recovering such information, as we do for example with `TILE_FORMATS`, as this column name feels less "common" than this one?

Reply:
Done